src/main/java/frc/trigon/robot/subsystems/arm/Arm.java

```diff
@@ -0,0 +1,138 @@
+        angleMotorProfile = new TrapezoidProfile(
+                ArmConstants.ANGLE_CONSTRAINTS,
+                new TrapezoidProfile.State(targetAngle.getDegrees(), 0),
+                new TrapezoidProfile.State(getAngleMotorPositionDegrees().getDegrees(), getAngleMotorVelocityInDegreesPerSeconds())
+        );
+        lastAngleMotorProfileGenerationTime = Timer.getFPGATimestamp();
+    }
+
+    private void generateElevatorMotorProfile(double targetElevatorPosition) {
+        elevatorMotorProfile = new TrapezoidProfile(
+                ArmConstants.ELEVATOR_CONSTRAINTS,
+                new TrapezoidProfile.State(targetElevatorPosition, 0),
+                new TrapezoidProfile.State(getElevatorMotorPositionRevolutions(), getElevatorMotorVelocityInRevolutionsPerSecond())
+        );
+        lastElevatorMotorProfileGenerationTime = Timer.getFPGATimestamp();
+    }
+
+    private void setTargetAngleFromProfile() {
+        if (angleMotorProfile == null) {
+            stopAngleMotors();
+            return;
+        }
+        final TrapezoidProfile.State targetState = angleMotorProfile.calculate(getAngleMotorProfileTime());
+        ArmConstants.MASTER_ANGLE_MOTOR.getPIDController().setP(ArmConstants.ANGLE_PID_CONTROLLER.getP());
+        ArmConstants.MASTER_ANGLE_MOTOR.getPIDController().setI(ArmConstants.ANGLE_PID_CONTROLLER.getI());
+        ArmConstants.MASTER_ANGLE_MOTOR.getPIDController().setD(ArmConstants.ANGLE_PID_CONTROLLER.getD());
+    }
+
+    private void setTargetPositionFromProfile() {
+        if (elevatorMotorProfile == null) {
+            stopElevatorMotors();
+            return;
+        }
+        final TrapezoidProfile.State targetState = elevatorMotorProfile.calculate(getElevatorMotorProfileTime());
+        ArmConstants.MASTER_ELEVATOR_MOTOR.getPIDController().setP(ArmConstants.ELEVATOR_PID_CONTROLLER.getP());
+        ArmConstants.MASTER_ELEVATOR_MOTOR.getPIDController().setI(ArmConstants.ELEVATOR_PID_CONTROLLER.getI());
+        ArmConstants.MASTER_ELEVATOR_MOTOR.getPIDController().setD(ArmConstants.ELEVATOR_PID_CONTROLLER.getD());
+    }
+
+    private double getAngleMotorProfileTime() {
+        return Timer.getFPGATimestamp() - lastAngleMotorProfileGenerationTime;
+    }
+
+    private double getElevatorMotorProfileTime() {
+        return Timer.getFPGATimestamp() - lastElevatorMotorProfileGenerationTime;
+    }
+
+    private Rotation2d getAngleMotorPositionDegrees() {
+        return Rotation2d.fromRotations(angleEncoder.getPosition().getValue());
+    }
+
+    private double getElevatorMotorPositionRevolutions() {
+        return magTicksToRevolutions(elevatorEncoder.getSelectedSensorPosition());
+    }
+
+    private double getAngleMotorVelocityInDegreesPerSeconds() {
+        return angleEncoder.getVelocity().getValue();
+    }
+
+    private double getElevatorMotorVelocityInRevolutionsPerSecond() {
+        return magTicksToRevolutions(elevatorEncoder.getSelectedSensorVelocity());
```

Comment:
This doesn't return in seconds

Reply:
How do I make it return in seconds?

Reply:
Check in which unit it currently returns, and use a function from `Conversions`

Reply:
I do but I can't figure out how to make it in seconds

Reply:
use a function from Conversions